src/index.ts

```diff
@@ -860,7 +860,9 @@ export class ValidationService {
 
                 // Firefox fix: redispatch 'submit' after finished handling this event
                 await new Promise(resolve => setTimeout(resolve, 0));
-                this.handleValidated(form, success, e);
+                if (e) {
+                    this.handleValidated(form, success, e);
```

Comment:
I _think_ it makes more sense to pass `e` through here and ignore it in `handleValidated`, e.g.

```ts
    handleValidated = (form: HTMLFormElement, success: boolean, submitEvent?: SubmitEvent) => {
        if (success) {
            if (submitEvent) {
                this.submitValidForm(form, submitEvent);
            }
        }
        else {
            this.focusFirstInvalid(form);
        }
    }
```

That's a pretty small/straightforward method to override if someone needs different behavior for whatever reason.

Note I also adjusted `submitEvent?: SubmitEvent`. If we know we might pass around an `undefined` we should model that explicitly.

Reply:
In retrospect it probably makes more sense to have `preValidate(form: HTMLFormElement, submitEvent?: SubmitEvent)`, with the missing event handled there instead of inside the form callback. That's technically a breaking change for anyone who's overriding `preValidate()`, but that may have just been me? (cc #33)

Reply:
Yeah, moving the check inside `handleValidated` is definitely better. The way I did it would prevent `focusFirstInvalid` from getting called even though it doesn't depend on the `submitEvent`.